### PR TITLE
Added support to 'loc-key' and 'loc-args' parameters to APNS.

### PR DESCRIPTION
**Pull request by godness84**
https://issues.jboss.org/browse/AGPUSH-1563

:dart: Targets: `master` and `1.1.x`

### CODE REVIEW

**Comment by lfryc** [comment]
Hey @godness84 , thanks for update!

I'd leave on @corinnekrych or @sebastienblanc to check the update itself as I'm not that familiar with java-apns  and APNs message format itself. For me it seems valid. :-)

However I can speak for code-style - I will comment on the files inline.

File: push/model/src/main/java/org/jboss/aerogear/unifiedpush/message/apns/APNs.java [hunk 1]
```diff
@@ -43,6 +43,12 @@
 
     @JsonProperty("content-available")
     private boolean contentAvailable;
+	
+    @JsonProperty("localized-key")
+    private String localizedKey;	
+	
+	@JsonProperty("localized-arguments")
```
Comment (lfryc) [inline push/model/src/main/java/org/jboss/aerogear/unifiedpush/message/apns/APNs.java]: wrong indentation

File: push/model/src/main/java/org/jboss/aerogear/unifiedpush/message/apns/APNs.java [hunk 2]
```diff
@@ -111,6 +117,11 @@ public void setContentAvailable(boolean contentAvailable) {
...
     public void setLocalizedTitleArguments(String[] localizedTitleArguments) {this.localizedTitleArguments = localizedTitleArguments;}
 
+	public String[] getLocalizedArguments() { return localizedArguments; }
+	
+	public void setLocalizedArguments(String[] localizedArguments){ this.localizedArguments = localizedArguments; }
 
-
+	public String getLocalizedKey() { return localizedKey; }
+	
+	public void setLocalizedKey(String localizedKey) { this.localizedKey = localizedKey; }
```
Comment (lfryc) [inline push/model/src/main/java/org/jboss/aerogear/unifiedpush/message/apns/APNs.java]: again, wrong indentation + please avoid one-liners (keeping the same formatting as the code around is usually best approach)

**Comment by lfryc** [comment]
As you update the formatting and placement of the new arguments, could you please also add a doc update here: https://github.com/godness84/aerogear-unifiedpush-server/blob/master/push/model/src/main/java/org/jboss/aerogear/unifiedpush/message/UnifiedPushMessage.java#L60-L61

and the test needs to be fixed as you can see from Travis build log: https://api.travis-ci.org/jobs/101864583/log.txt?deansi=true

**Comment by lfryc** [comment]
Could you please also create an issue in [JIRA](http://issues.jboss.org/browse/AGPUSH) and link it back here in Pull Request description?

**Comment by lfryc** [comment]
@godness84 thanks for lot of updates, but consistency is a good thing :smiley:

**Comment by matzew** [comment]
Lukas, see AGPUSH-1563



On Tuesday, 12 January 2016, Lukáš Fryč <notifications@github.com> wrote:

> @godness84 <https://github.com/godness84> thanks for lot of updates, but
> consistency is a good thing [image: :smiley:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear-unifiedpush-server/pull/669#issuecomment-170957158>
> .
>


-- 
Sent from Gmail Mobile

**Comment by godness84** [comment]
Yes, you're right, indentation is wrong, maybe my editor messed up.
As @matzew pointed out, the issue in JIRA has been already created and linked back.
ASAP I will do what you asked me :)

**Comment by lfryc** [comment]
Perfect, thanks @godness84 !

**Comment by godness84** [comment]
@lfryc, I did my homeworks :) Let me know if there is still something wrong

**Comment by lfryc** [comment]
:+1: seems great now! thanks @godness84

**Comment by lfryc** [comment]
Merging...

**Comment by lfryc** [comment]
Thanks. This pull request has been integrated manually.

**Comment by godness84** [comment]
Great, thanks @lfryc !

**Comment by lfryc** [comment]
:rocket: landed `master`: https://github.com/aerogear/aerogear-unifiedpush-server/commit/8c19aaec50f3fc2c3ca4f3cef453dffc59e7a1f6

:rocket: landed `1.1.x`: https://github.com/aerogear/aerogear-unifiedpush-server/commit/c368c070ef7ed854b4b562fc906b975790cbd8e5

**Pull request closed without merging**